### ISSUE TITLE
Beam's pull request to use Libraries BOM 16.3.0 had dependency conflicts

### ISSUE THREAD

**Issue by suztomo** [GoogleCloudPlatform/cloud-opensource-java#1930]
https://github.com/apache/beam/pull/13765#issuecomment-768785230

The pull request is trying to upgrade the libraries-bom version to 16.3.0 from 13.2.0.

- the libraries-bom 16.3.0 has "google-cloud-bigquerystorage:1.8.0" which has "grpc-api:1.34.1" dependency.
- the libraries-bom 16.3.0 has "grpc-core:1.35.0".
- "grpc-api:1.34.1" is not compatible with "grpc-core:1.35.0".

A Beam user (Maven, without using the upcoming Beam GCP BOM) would have the dependency conflict.

@elharo  Can we do something about such discrepancy within the member of the libraries BOM?

**Comment by elharo** [comment]
I don't think we can. What is the incompatibility? Is gRPC 1.35 not backwards compatible with 1.34.1 or is there some reason beam gets 1.34.1 instead of 1.35.0?

**Comment by suztomo** [comment]
@elharo The incompatibility is that "grpc-core:1.35.0" references newly-added methods (e.g., `InternalConfigSelector$Result.getInterceptor()`, `LoadBalancer$Helper.createResolvingOobChannelBuilder(String, ChannelCredentials)`, `LoadBalancer$Helper.getChannelCredentials()`, etc.) that are not available in the old "grpc-api:1.34.1". More details in https://gist.github.com/suztomo/6495cc692e09deb2327798f6f6a3261d#file-gistfile1-txt-L1007.

>  is there some reason beam gets 1.34.1 instead of 1.35.0?

Yes, this is the case. When a Maven user declares "org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.29.0-SNAPSHOT" dependency (without using the libraries BOM / Beam GCP BOM), Maven's dependency resolution would select the old "grpc-api:**1.34.1**":

```
io.grpc:grpc-api:1.34.1 is at:
  org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.29.0-SNAPSHOT / com.google.cloud:google-cloud-bigquerystorage:1.8.0 (compile) / io.grpc:grpc-api:1.34.1 (compile)
...
io.grpc:grpc-core:1.35.0 is at:
  org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.29.0-SNAPSHOT / io.grpc:grpc-core:1.35.0 (compile)
```

In other words, the `org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.29.0-SNAPSHOT` would have dependency conflicts in its own dependencies.

**Comment by suztomo** [comment]
beam-sdks-java-io-google-cloud-platform module should declare grpc-api. https://github.com/apache/beam/pull/13765#issuecomment-769933351
